### PR TITLE
Update task/feature names

### PR DESCRIPTION
**Pull request by pcuenca**
According to the nomenclature used in the Hub https://huggingface.co/tasks, and bringing labels in line with https://github.com/huggingface/optimum/pull/967, which was addressed after a Slack discussion.

### CODE REVIEW

File: src/exporters/coreml/__main__.py [hunk 1]
```diff
@@ -74,8 +76,8 @@ def main():
     )
     parser.add_argument(
         "--feature",
-        choices=list(FeaturesManager.AVAILABLE_FEATURES),
-        default="default",
+        choices=list(FeaturesManager.AVAILABLE_FEATURES_INCLUDING_LEGACY),
```
Comment (pcuenca) [inline src/exporters/coreml/__main__.py:79]: I arbitrarily decided to deprecate the old names but still support them in the CLI (not the API). Happy to remove all support at once, or allow the legacy names to still be used elsewhere.

**Review by hollance — APPROVED** [review]
I think this is fine. The goal has been to eventually move this into Optimum anyway, so anything that goes in that direction is good.

File: src/exporters/coreml/__main__.py [hunk 2]
```diff
@@ -121,14 +123,21 @@ def main():
...
+        deprecation_message = f"Feature '{feature}' is deprecated, please use '{args.feature}' instead."
+        warnings.warn(deprecation_message, FutureWarning)
 
     # Allocate the model
     model = FeaturesManager.get_model_from_feature(
         args.feature, args.model, framework=args.framework, #cache_dir=args.cache_dir
     )
     model_kind, model_coreml_config = FeaturesManager.check_supported_model_or_raise(model, feature=args.feature)
 
-    if args.feature in ["seq2seq-lm", "speech-seq2seq"]:
```
Comment (fxmarty) [inline src/exporters/coreml/__main__.py:131]: For `speech-seq2seq`, I renamed it as well to `automatic-speech-recognition` to align with https://huggingface.co/datasets/huggingface/transformers-metadata/blob/main/pipeline_tags.json

**Review by fxmarty — APPROVED** [review]
LGTM!

File: src/exporters/coreml/__main__.py (hunk 2 above)
Comment (pcuenca) [reply src/exporters/coreml/__main__.py:131]: Yes, I saw that but found it a bit confusing to deal with that single special case here. I'd rather keep them separate for simplicity and we'll address the change when we move to Optimum, if that's ok.

I did change some inconsistent use of both `"speech-recognition"` and `"automatic-speech-recognition"` in the previous commit.

**Pull request merged**